cdragontoolbox/binfile.py

```diff
@@ -361,6 +414,33 @@ def read_binfile_entry(self, htype):
         assert self.f.tell() - pos == length
         return entry
 
+    def read_patch_section(self):
+        """Reads the entire PTCH section"""
+
+        count = self.read_u32()
+        patch_entries = {}
+        for _ in range(count):
+            hpath = self.read_fmt('<2L')[0]
+            btype = self.parse_bintype(self.read_u8())
+            objectpath = self.read_string()
+            parts = objectpath.split('.')
+            binvalue = self.read_field_basic(compute_binhash(parts[-1]), btype)
+            if hpath not in patch_entries:
+                if len(parts) > 1:
+                    patch_entries[hpath] = BinPtchEntry(hpath, [BinNestedField(compute_binhash(parts[0]), BinNested([]))])
+                else:
+                    patch_entries[hpath] = BinPtchEntry(hpath, [])
```

Comment:
The special case of the initialization can be avoided by using a `BinNested` as the top level (directly in the `BinPtchEntry`).
`BinPtchEntry` would be adjusted to simply contain two fields: `hpath` (same as current), and `value` (which would be a `BinNested`).
It should make code cleaner since the whole patch/nesting stuff will be only solely by `BinNested` and its friend `BinNestedField` (no need for special processing in `BinPtchEntry`).
Note that the same could probably be made for `BinEntry`, if it stored a `BinStruct` directly instead of a list of fields; it's outside of the scope of this PR though so don't worry about it.

For patch entries this would look like this.
```py
class BinPtchEntry:
    def __init__(self, hpath, value):
        self.path = BinEntryPath(hpath)
        self.value = value  # BinNested
    ...
    # Note: to_serializable() needs adjustements too


def read_patch_section(self):
    ...

    if hpath not in patch_entries:
        patch_entries[hpath] = BinPtchEntry(hpath, BinNested([]))
    current_nesting = patch_entries[hpath].value

    for part in range(0, len(parts)-1):
        if part not in current_nesting:
            current_nesting[part] = BinNestedField(compute_binhash(part), BinNested([]))
        current_nesting = current_nesting[part]

    current_nesting[parts[-1]] = binvalue
```

Reply:
I don't understand.